app/src/org/commcare/android/resource/installers/XFormAndroidInstaller.java

```diff
@@ -74,19 +74,21 @@ public boolean initialize(AndroidCommCarePlatform platform, boolean isUpgrade) t
             IOException, InvalidReferenceException, InvalidStructureException,
             XmlPullParserException, UnfullfilledRequirementsException {
         super.initialize(platform, isUpgrade);
-        if (isLatestFormRecord(platform)) {
+
+        if (platform.getFormDefId(namespace) == -1) {
             platform.registerXmlns(namespace, formDefId);
+        } else {
+            // Only overwrites the existing form if the resourceVersion of the new form is higher than the existing one
+            FormDefRecord existingForm = FormDefRecord.getFormDef(platform.getFormDefStorage(), platform.getFormDefId(namespace));
+            FormDefRecord newForm = FormDefRecord.getFormDef(platform.getFormDefStorage(), formDefId);
+            if (newForm.getResourceVersion() > existingForm.getResourceVersion()) {

```

Comment:
Small point that I think this should be `>=`. 

There could be a situation where the old resource and new resource have the same version and the old resource is on the way out. In that case the signal that this current resource is installed (because it is being initialized, which only happens to installed resources) should imply that this resource is valid, so with no other information about the installed resource we should presume that it may not be.